### PR TITLE
62 improvement of the action reaction services

### LINKED ISSUE #62: improvement of the action & reaction services
## Improvement
refactoring of functions to make them generic.
creation of triggers for all action services.

### PR DESCRIPTION
**Pull request by STom6**
# Updated feature
implementation of all triggers for action services:
Lower / upper for iss position.
Good weather / bad weather for the openMeteo service

### CODE REVIEW

**Review by JohanCDev — COMMENTED** [review]
Lgtm

**Review by CedricCORGE — DISMISSED** [review]
lgtm

File: Server/Services/openMeteoService.js [hunk 1]
```diff
@@ -36,42 +36,56 @@ let comparaisons = [
...
-                            if (comparaison.result == weatherCode) {
-                                console.log('success, send an email')
-                                googleService.send_mail(`Le temps est : ${comparaison.name} le ${date} à ${hour} heures`, `météo a ${hour} heures`, 'AREA METEO', uid)
-                                return;
-                            }
-                        });
-                    }
-                });
+    /**
+    * @brief Check if it is weather is fine or not based on the current date and
```
Comment (osvegn) [inline Server/Services/openMeteoService.js]: > Check if the wheather is fine or not based on the current ...

File: Server/Services/openMeteoService.js [hunk 2]
```diff
@@ -36,42 +36,56 @@ let comparaisons = [
...
-        })
-        .catch(error => {
-            console.log(error);
+        });
+    },
+
+    /**
+    * @brief GetLocation function retrieves the location data from Firebase for the specified user ID.
+    * @param uid (string) user ID
+    * @returns Promise that resolves with location data if successful or rejects with an error if unsuccessful
```
Comment (osvegn) [inline Server/Services/openMeteoService.js]: The sentence may be improved, I don't understand

**Review by osvegn — CHANGES_REQUESTED** [review]
Improve the documentation

**Review by 31Nathan — APPROVED** [review]
LGTM

**Review by JohanCDev — APPROVED** [review]
lgtm

**Review by osvegn — APPROVED** [review]
Look good

**Pull request merged**